include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -11,33 +11,116 @@
 #define _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_COMMON_H
 
 #include <sycl/sycl.hpp>
+#include <atomic>
 #include <cstddef>
 #include <cstdlib>
 #include <cassert>
 #include <cstring>
 #include <limits>
 
 #if __linux__
-#include <dlfcn.h>
-#include <unistd.h>
+#    include <dlfcn.h>
+#    include <unistd.h>
 #endif // __linux__
 
 namespace __pstl_offload
 {
 
+#if __linux__
+inline void*
+__original_aligned_alloc(std::size_t __alignment, std::size_t __size)
+{
+    using __aligned_alloc_func_type = void* (*)(std::size_t, std::size_t);
+
+    static __aligned_alloc_func_type __orig_aligned_alloc =
+        __aligned_alloc_func_type(dlsym(RTLD_NEXT, "aligned_alloc"));
+    return __orig_aligned_alloc(__alignment, __size);
+}
+#endif // __linux__
```

Comment:
Is it needed to move this function to here? It does not seem to be used in this header.

Reply:
Yes, this is run away from an experiment. Moved back.